### ISSUE TITLE
ConfigPhase.BUILD_AND_RUN_TIME_FIXED allows to set properties in JVM mode

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#17614]
## Describe the bug

A user [has reported](https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/http.20permission.20enabled.20runtime.20or.20buildtime.20property) that [enabling the HTTP permissions](https://quarkus.io/guides/security-authorization#disabling-permissions) is allowed in the JVM mode but not in the native mode:

`HttpBuildTimeConfig` -> `AuthConfig.permissions` -> `PolicyMappingConfig.enabled`

`HttpBuildTimeConfig` is a `ConfigRoot` in a `ConfigPhase.BUILD_AND_RUN_TIME_FIXED` so it explains why this property is shown with a lock on the all config page.

### Expected behavior
Setting properties at runtime should not work in `ConfigPhase.BUILD_AND_RUN_TIME_FIXED`.

`HttpBuildTimeConfig` should be split into `HttpBuildTimeConfig` which would only keep the `ConfigPhase.BUILD_AND_RUN_TIME_FIXED` properties and `HttpConfig` which would have all other properties which may be possible to override at runtime (if it is possible in principle for some HTTP properties)

**Comment by radcortez** [comment]
Yes, this is related with https://github.com/quarkusio/quarkus/pull/16101.

The main issue is that some extensions rely on configuration that is not recorded during build time in the defaults, so they need the sources anyway to read the configuration (which has the side effect to allow for an override). This was always like this, but in https://github.com/quarkusio/quarkus/pull/16101 we were able to at least remove the user sources from the equation. We require some more in depth work to proceed and remove everything else. 

This https://github.com/quarkusio/quarkus/pull/17228 should be another step into the right direction, because it will allow configurations outside of the `quarkus` namespace to be recorded and later on remove the default sources during static init to keep only the defaults one (generated at build time).
